crates/shared/src/trade_finding.rs

```diff
@@ -161,11 +161,13 @@ impl Trade {
             OrderKind::Sell => order_amount
                 .mul(&sell_price)
                 .checked_div(&buy_price)
-                .context("div by zero: buy price")?,
+                .context("div by zero: buy price")?
+                .ceil(),
             OrderKind::Buy => order_amount
                 .mul(&buy_price)
                 .checked_div(&sell_price)
-                .context("div by zero: sell price")?,
+                .context("div by zero: sell price")?
+                .ceil(),
```

Comment:
We only need `.ceil()` to compute the buy_amount (i.e. out_amount) of sell orders. See [here](https://github.com/cowprotocol/contracts/blob/main/src/contracts/GPv2Settlement.sol#L389-L411).